### PR TITLE
Feature/add tests

### PR DESCRIPTION
**Pull request by christophe-chausseray**

### CODE REVIEW

File: Tests/Render/FrontRenderTest.php [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    {
+        $this->frontRender = new FrontRender($this->engine, $this->eventDispatcher, self::TEMPLATE);
+
+        $this->frontRender->setParameters(
+            [
+                'applicationName' => 'Test Render',
+            ]
+        );
+
+        $response = new Response($this->frontRender->render());
```
Comment (Neirda24) [inline Tests/Render/FrontRenderTest.php]: No need for the response object.

File: Tests/Render/FrontRenderTest.php [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    }
+
+    /**
+     * Test render of the template without parameter
+     */
+    public function testTemplateRenderWithoutParameter()
+    {
+        $this->frontRender = new FrontRender($this->engine, $this->eventDispatcher, self::TEMPLATE);
+
+        $response = new Response($this->frontRender->render());
```
Comment (Neirda24) [inline Tests/Render/FrontRenderTest.php]: same

File: Tests/Render/FrontRenderTest.php [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+            $response->getContent()
+        );
+    }
+
+    /**
+     * Test on the update of lexer
+     *
+     * @dataProvider lexerTags
+     *
+     * @param $oldTags
```
Comment (Neirda24) [inline Tests/Render/FrontRenderTest.php]: type

File: Tests/Render/FrontRenderTest.php [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+    {
+        $this->frontRender = new FrontRender($this->engine, $this->eventDispatcher, self::TEMPLATE);
+
+        $this->frontRender->setParameters(
+            [
+                'applicationName' => 'Test Render',
+            ]
+        );
+
+        $response = new Response($this->frontRender->render());
```
Comment (Neirda24) [inline Tests/Render/FrontRenderTest.php]: no need

File: Tests/Render/FrontRenderTest.php [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
+        $this->assertEquals(200, $response->getStatusCode());
```
Comment (Neirda24) [inline Tests/Render/FrontRenderTest.php]: no need

File: Tests/Render/FrontRenderTest.php [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+    public function lexerTags()
+    {
+        return [
+            ['{#', '{*'],
+            ['{%', '{@'],
+            ['{{', '{\$']
+        ];
+
+    }
+
```
Comment (Neirda24) [inline Tests/Render/FrontRenderTest.php]: remove blank lines

File: phpunit.xml.dist [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+         colors="true"
+         bootstrap="vendor/autoload.php"
+>
+    <php>
+        <ini name="error_reporting" value="-1" />
+        <server name="KERNEL_DIR" value="/home/ironweb/iron/front-render-bundle/" />
+    </php>
+
+    <testsuites>
+        <testsuite name="Project Test Suite">
```
Comment (kshitizlondon) [inline phpunit.xml.dist]: Test suite name without the spaces is preferred.

File: phpunit.xml.dist [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+            <directory>./Tests</directory>
+        </testsuite>
+        <testsuite name="Vendor Test">
```
Comment (kshitizlondon) [inline phpunit.xml.dist]: Test suite name without the spaces is preferred.

**Comment by kshitizlondon** [comment]
:+1:

**Pull request merged**